src/main/java/io/github/nucleuspowered/nucleus/modules/ticket/handlers/TicketHandler.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.modules.ticket.handlers;
+
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.RemovalListener;
```

Comment:
In these scenarioes, though, I don't think the performance difference would have a huge effect. In general, I would rather be able to minimise dependencies and I'm not keen on adding unneeded dependencies for the sake of. I guess 3.5MB (which is the current Nucleus size) isn't exactly large in today's terms, but I'm wary of adding to the dependency tree unnecessarily - which I feel this is potentially erring on. This is why I'm questioning it.

I'll have to look into it more, though for the sake of a 145kB library that doesn't have any other required deps for runtime, it's not the biggest problem in the world, I guess. I do  remember the Sponge guys removing Caffeine from the dependencies - though I suspect that might just have been lack of use. 

I suspect the upshot here is the handling of futures, really...